src/assets/wise5/directives/teacher-summary-display/ideas-summary.component.ts

```diff
@@ -0,0 +1,201 @@
+      } else {
+        this.leastCommonIdeas = [...sortedIdeas]
+          .splice(sortedIdeas.length - 3, sortedIdeas.length)
+          .reverse();
+      }
+      this.allIdeas = this.sortIdeasById(ideaCountArray);
+    }
+  }
+
+  private ideaCountMapToArray(ideaDescriptions: CRaterIdea[]): IdeaData[] {
+    const ideaCountArray = [];
+    this.ideaCountMap.forEach((count, ideaId, map) => {
+      const ideaDescription = ideaDescriptions.find(
+        (ideaDescription) => ideaDescription.name === ideaId
+      );
+      ideaCountArray.push({
+        id: ideaId,
+        text: this.useIdeaTextOrId(ideaId, ideaDescription?.text),
+        count: count
+      });
+    });
+    return ideaCountArray;
+  }
+
+  private useIdeaTextOrId(id: string, text: string): string {
+    return text ?? 'idea ' + id;
+  }
+
+  private sortIdeasByCount(ideas: IdeaData[]): IdeaData[] {
+    return ideas.filter((idea) => idea.count > 0).sort((a, b) => b.count - a.count);
+  }
+
+  private sortIdeasById(ideas: IdeaData[]): IdeaData[] {
+    let sorted = ideas
+      .filter((idea) => !this.stringContainsLetters(idea.id))
+      .sort((a, b) => Number(a.id) - Number(b.id));
+
+    const sortedIdeasWithLetters = this.getSortedIdeasWithLetters(ideas);
+
+    return this.insertIdeasWithLetters(sorted, sortedIdeasWithLetters);
+  }
+
+  private stringContainsLetters(str: string): boolean {
+    let hasLetters = false;
+    Array.from(str).forEach((char) => {
+      if (isNaN(Number(char))) {
+        hasLetters = true;
+      }
+    });
+    return hasLetters;
+  }
+
+  private getSortedIdeasWithLetters(ideas: IdeaData[]) {
+    return ideas
+      .filter((idea) => this.stringContainsLetters(idea.id))
+      .sort((a, b) => {
+        const prefixDif = this.stringNumericPrefix(a.id) - this.stringNumericPrefix(b.id);
+        if (prefixDif === 0) {
+          return a.id.localeCompare(b.id);
+        } else {
+          return prefixDif;
+        }
+      });
+  }
+
+  private insertIdeasWithLetters(
+    sorted: IdeaData[],
+    sortedIdeasWithLetters: IdeaData[]
+  ): IdeaData[] {
+    for (let i = 0; i < sorted.length; i++) {
+      while (
+        sortedIdeasWithLetters.length > 0 &&
+        Number(sorted.at(i).id) > this.stringNumericPrefix(sortedIdeasWithLetters.at(0).id)
+      ) {
+        const ideaWithLetter = sortedIdeasWithLetters.at(0);
+        sortedIdeasWithLetters = sortedIdeasWithLetters.slice(1, sortedIdeasWithLetters.length);
+        sorted.splice(i, 0, ideaWithLetter);
+        i++;
+      }
+    }
+    return sorted;
+  }
+
+  private stringNumericPrefix(str: string): number {
+    let numericPrefix = '';
+    const strArray = Array.from(str);
+    for (let charIndex = 0; charIndex < strArray.length; charIndex++) {
+      const char = strArray.at(charIndex);
+      if (isNaN(Number(char))) {
+        break;
+      } else {
+        numericPrefix = numericPrefix.concat(char);
+      }
+    }
+    return Number(numericPrefix);
+  }
```

Comment:
Do these functions need to be defined in this component? Put these functions in a new class, and write tests?